src/app/components/work-item-detail/work-item-detail.component.ts

```diff
@@ -1003,26 +932,46 @@ export class WorkItemDetailComponent implements OnInit, AfterViewInit, OnDestroy
     return result;
   }
 
-  findIterationById(iterationId: string): IterationModel {
-    for (let i=0; i<this.iterations.length; i++)
-      if (this.iterations[i].id === iterationId)
-        return this.iterations[i];
-    return null;
-  }
+  // findIterationById(iterationId: string): IterationModel {
+  //   for (let i=0; i<this.iterations.length; i++)
+  //     if (this.iterations[i].id === iterationId)
+  //       return this.iterations[i];
+  //   return null;
+  // }
 
   focusArea() {
     this.iterationSelectbox.close();
     this.cancelAssignment();
+    this.areas = [
+      ...this.areas,
+      {
+        key: '0',
+        value: '',
+        selected: false,
+        cssLabelClass: 'spinner spinner-sm spinner-inline'
+      }
+    ];
+    this.getAreas();
   }
 
   focusIteration() {
     this.areaSelectbox.close();
     this.cancelAssignment();
+    this.iterations = [
+      ...this.iterations,
+      {
+        key: '0',
+        value: '',
+        selected: false,
+        cssLabelClass: 'spinner spinner-sm spinner-inline'
+      }
+    ];
+    this.getIterations();
   }
 
   areaUpdated(areaId: string) {
-
```

Comment:
We need to emit the emitEditWI event in the areaUpdated function as save function is no longer emitting it.